connectors/filtering/validation.py

```diff
@@ -118,6 +124,20 @@ def to_s(cls, value):
             case FilteringValidationState.EDITED:
                 return "edited"
 
+    @classmethod
+    def from_string(cls, string):
+        match string.casefold():
+            case "valid":
+                return FilteringValidationState.VALID
+            case "invalid":
+                return FilteringValidationState.INVALID
+            case "edited":
+                return FilteringValidationState.EDITED
+            case _:
+                raise InvalidFilteringValidationStateError(
+                    f"'{string}' is an unknown value for the enum FilteringValidationState. Allowed states: {FilteringValidationState.STATES}.)"
```

Comment:
I would remove `STATE` and just call `[e.value for e in FilteringValidationState]` here

Reply:
Let's use the `str2e` function in #449 . The function assumes `None` or a valid enum value, and ignore invalid values, which I think it's good enough for now.

Let's keep the processing of Enum in a single place and we can enhance later.